### PR TITLE
feat(db)!: use accumulating queries

### LINKED ISSUE #915: `fill-analytics` should use incremental analytics computation
There are still some analytics queries that take a relatively long time. We should think about making these queries incremental, i.e. we get an initial state and derive the absolute values of the next milestones by applying deltas.

### PR DESCRIPTION
**Pull request by Alexandcoats**
## Linked Issues

* Closes #915

## Notes to Reviewer

As a reviewer, please pay particular attention to the following areas when reviewing this PR and tick the above boxes after you have completed the steps.

#### Test cases
* [x] Suggest places that may benefit from test cases.

#### INX Changes
* [x] Run chronicle using an INX connection.

#### Database Changes
* [x] Review database queries for correctness/conciseness.
* [x] Ensure queries are supported by indexes if needed.
* [x] Check for breaking changes in the data model and matching (conventional) commit message prefix.

### CODE REVIEW

**Comment by Alex6323** [comment]
```
2022-11-24T17:07:10.489702Z  INFO inx_chronicle::cli: Finished analytics for milestone 987785
2022-11-24T17:07:10.904328Z  INFO inx_chronicle::cli: Finished analytics for milestone 970415
2022-11-24T17:07:10.912520Z ERROR chronicle::db::collections::outputs::analytics: error=invalid value: integer `-1`, expected u64
2022-11-24T17:07:10.913735Z ERROR chronicle::db::collections::analytics: error=invalid value: integer `-1`, expected u64
Error: invalid value: integer `-1`, expected u64
```

File: src/db/collections/outputs/mod.rs [hunk 1]
```diff
@@ -487,6 +487,133 @@ mod analytics {
...
+        ) -> Result<(), Error> {
+            #[derive(Default, Deserialize)]
+            struct Sums {
+                count: u64,
+                value: d128,
+            }
+
+            #[derive(Default, Deserialize)]
+            #[serde(default)]
+            struct Res {
```
Comment (grtlr) [inline src/db/collections/outputs/mod.rs:507]: We use `Res` in multiple places, why can't it live outside of the function?

File: src/db/collections/outputs/mod.rs [hunk 2]
```diff
@@ -514,7 +641,12 @@ mod analytics {
                                 } },
                             ],
                             "nft_changed": [
-                                { "$match": { "output.nft_id": { "$ne": NftId::implicit() } } },
+                                { "$match": { 
+                                    "$and": [
+                                        { "output.nft_id": { "$exists": true } },
```
Comment (grtlr) [inline src/db/collections/outputs/mod.rs:646]: This means that our previous computation was wrong, right? I've been seeing fishy numbers in the Grafana dashboard :grimacing:.

File: src/db/collections/outputs/mod.rs [hunk 3]
```diff
@@ -615,6 +750,75 @@ mod analytics {
...
+        ) -> Result<(), Error> {
+            let (created, consumed) = tokio::try_join!(
+                async {
+                    Result::<_, Error>::Ok(self.aggregate::<LedgerSizeAnalytics>(
+                        vec![
+                            doc! { "$match": {
+                                "metadata.booked.milestone_index": ledger_index,
+                                "metadata.spent_metadata.spent.milestone_index": { "$ne": ledger_index }
+                            } },
+                            doc! { "$group" : {
```
Comment (grtlr) [inline src/db/collections/outputs/mod.rs:770]: I really think we should get rid of the redundancy between the pipeline stages. There is high risk for bugs if the two queries (accidentally) diverge.

Comment (Alex6323) [reply src/db/collections/outputs/mod.rs:729]: Not in this PR though, right?

File: src/db/collections/outputs/mod.rs [hunk 4]
```diff
@@ -754,20 +909,52 @@ mod analytics {
...
+                            "metadata.booked.milestone_index": { "$eq": 0 },
+                            "metadata.spent_metadata.spent.milestone_index": ledger_index
+                        } },
+                        doc! { "$group": {
+                            "_id": null,
+                            "unclaimed_count": { "$sum": 1 },
+                            "unclaimed_value": { "$sum": { "$toDecimal": "$output.amount" } },
+                        } },
+                        doc! { "$project": {
+                            "unclaimed_count": 1,
```
Comment (Alex6323) [inline src/db/collections/outputs/mod.rs:934]: Why is this set to `1`? @grtlr

Comment (Alex6323) [reply src/db/collections/outputs/mod.rs:934]: yeah, that must be it...didn't think of that.

**Comment by Alex6323** [comment]
LGTM!

**Comment by grtlr** [comment]
I'm closing this one as well in favor of #1061.

**Pull request closed without merging**